### PR TITLE
Forbidden to add ns __namespace

### PR DESCRIPTION
**Pull request by mathspanda**
Fix issue #770 
1. Forbidden to create namespace "__namespace"
2. Add default namespace "__namespace" into the results of "namespace get *"

### CODE REVIEW

File: src/config.cc [hunk 1]
```diff
@@ -742,6 +742,9 @@ Status Config::AddNamespace(const std::string &ns, const std::string &token) {
   if (cluster_enabled) {
     return Status(Status::NotOK, "forbidden to add namespace when cluster mode was enabled");
   }
+  if (ns == kDefaultNamespace) {
+    return Status(Status::NotOK, "default namespace already exists");
```
Comment (git-hulk) [inline src/config.cc]:
```suggestion
    return Status(Status::NotOK, "forbidden to the add the default namespace");
```

File: tests/cppunit/config_test.cc [hunk 2]
```diff
@@ -179,6 +179,10 @@ TEST(Namespace, Add) {
   auto s = config.AddNamespace("n1", "t0");
   EXPECT_FALSE(s.IsOK());
   EXPECT_EQ(s.Msg(), "the namespace has already exists");
+
+  s = config.AddNamespace(kDefaultNamespace, "tdefault");
```
Comment (git-hulk) [inline tests/cppunit/config_test.cc]:
```suggestion
  s = config.AddNamespace(kDefaultNamespace, "mytoken");
```

File: tests/cppunit/config_test.cc [hunk 3]
```diff
@@ -179,6 +179,10 @@ TEST(Namespace, Add) {
...
+  EXPECT_FALSE(s.IsOK());
+  EXPECT_EQ(s.Msg(), "default namespace already exists");
```
Comment (git-hulk) [inline tests/cppunit/config_test.cc]:
```suggestion
  EXPECT_EQ(s.Msg(), "forbidden to the add the default namespace");
```

**Comment by git-hulk** [comment]
@mathspanda Thanks for finding out and solving this issue, left a few suggestions. You can commit them if look good to you.

File: src/config.cc [hunk 4]
```diff
@@ -742,6 +742,9 @@ Status Config::AddNamespace(const std::string &ns, const std::string &token) {
   if (cluster_enabled) {
     return Status(Status::NotOK, "forbidden to add namespace when cluster mode was enabled");
   }
+  if (ns == kDefaultNamespace) {
+    return Status(Status::NotOK, "forbidden to the add the default namespace");
```
Comment (git-hulk) [inline src/config.cc]:
```suggestion
    return Status(Status::NotOK, "forbidden to add the default namespace");
```

File: tests/cppunit/config_test.cc [hunk 5]
```diff
@@ -179,6 +179,10 @@ TEST(Namespace, Add) {
   auto s = config.AddNamespace("n1", "t0");
   EXPECT_FALSE(s.IsOK());
   EXPECT_EQ(s.Msg(), "the namespace has already exists");
+
+  s = config.AddNamespace(kDefaultNamespace, "mytoken");
+  EXPECT_FALSE(s.IsOK());
+  EXPECT_EQ(s.Msg(), "forbidden to the add the default namespace");
```
Comment (git-hulk) [inline tests/cppunit/config_test.cc]:
```suggestion
  EXPECT_EQ(s.Msg(), "forbidden to add the default namespace");
```

**Review by git-hulk — APPROVED** [review]

**Review by ShooterIT — APPROVED** [review]

**Comment by git-hulk** [comment]
Thanks all, merging...

**Pull request merged**